ExtendedControls/Forms/MessageBoxTheme.cs

```diff
@@ -15,152 +15,184 @@ public partial class MessageBoxTheme : DraggableForm
         {
             InitializeComponent();
+
+            DialogResult = DialogResult.None;
+
+            this.msgText = text;
+            this.Text = labelCaption.Text = caption ?? string.Empty;
+            this.buttons = buttons;
+            this.mbIcon = messageBoxIcon;
+            if (formIcon != null)
+                this.Icon = formIcon;
         }
 
-        public void Init(string ptext, string caption, MessageBoxButtons? buttons, MessageBoxIcon ic, System.Drawing.Icon windowicon)
+        protected override void OnFormClosed(FormClosedEventArgs e)
         {
-            if (buttons == null)
-            {
-                buttonExt1.Visible = false;
-                buttonExt2.Visible = false;
-                buttonExt3.Visible = false;
-            }
-            else if (buttons == MessageBoxButtons.AbortRetryIgnore)
-            {
-                buttonExt1.Tag = DialogResult.Ignore; buttonExt1.Text = "Ignore";
-                buttonExt2.Tag = DialogResult.Retry; buttonExt2.Text = "Retry";
-                buttonExt3.Tag = DialogResult.Abort; buttonExt3.Text = "Abort";
-            }
-            else if (buttons == MessageBoxButtons.OK)
-            {
-                buttonExt1.Tag = DialogResult.OK; buttonExt1.Text = "OK";
-                buttonExt2.Visible = false;
-                buttonExt3.Visible = false;
-                this.AcceptButton = this.CancelButton = buttonExt1;
-            }
-            else if (buttons == MessageBoxButtons.OKCancel)
-            {
-                buttonExt1.Tag = DialogResult.Cancel; buttonExt1.Text = "Cancel";
-                buttonExt2.Tag = DialogResult.OK; buttonExt2.Text = "OK";
-                buttonExt3.Visible = false;
-            }
-            else if (buttons == MessageBoxButtons.RetryCancel)
-            {
-                buttonExt1.Tag = DialogResult.Cancel; buttonExt1.Text = "Cancel";
-                buttonExt2.Tag = DialogResult.OK; buttonExt2.Text = "Retry";
-                buttonExt3.Visible = false;
-            }
-            else if (buttons == MessageBoxButtons.YesNo)
+            base.OnFormClosed(e);
+            panelIcon?.Dispose();
+        }
+
+        protected override void OnLoad(EventArgs e)
+        {
+            base.OnLoad(e);
+
+            switch (buttons)
             {
-                buttonExt1.Tag = DialogResult.No; buttonExt1.Text = "No";
-                buttonExt2.Tag = DialogResult.Yes; buttonExt2.Text = "Yes";
-                buttonExt3.Visible = false;
+                case null:
+                    buttonExt1.Visible = buttonExt2.Visible = buttonExt3.Visible = false;
+                    break;
+                case MessageBoxButtons.AbortRetryIgnore:
+                    buttonExt1.DialogResult = DialogResult.Ignore; buttonExt1.Text = "&Ignore";
+                    buttonExt2.DialogResult = DialogResult.Retry; buttonExt2.Text = "&Retry";
```

Comment:
The ampersands are being picked up and tuning the next letter into the Alt+ shortcut key option in the Windows Default theme but this isn't working on any of the others and just includes the '&' in the text on the button.

Reply:
Sorry, the Alt+key does still work, it's just the hiding it in the display string and underlining the first character that doesn't happen on themes.

Reply:
I found a [decent article](https://www.cyotek.com/blog/adding-keyboard-accelerators-and-visual-cues-to-a-winforms-control) on winforms mnemonics and will l into getting this resolved when I'm back at my desk.